src/common/transport.cc

```diff
@@ -142,7 +142,7 @@ void Transport::handleIncoming(const std::shared_ptr<Peer> &peer) {
         if (errno == ECONNRESET) {
```

Comment:
Assuming the peer disconnects for all other errors is probably a good idea.  Throwing this exception is not useful.

However, please clean up the "if()" statement.  Just remove lines 142-146 and simply call "handlePeerDisconnection()" here.

Reply:
I adjusted it, please accept it